test/distributed/_spmd/test_data_parallel.py

```diff
@@ -165,56 +135,26 @@ def test_replicate_adam_fused(self):
     @skip_if_lt_x_gpu(2)
     @with_comms
     def test_fully_shard_sgd(self):
-        mod = SimpleMLP().cuda(self.rank)
-        opt = torch.optim.SGD(mod.parameters(), lr=0.1)
-
-        train_batch = (
-            torch.randn(128, 50).to(self.rank),
-            torch.randn(128, 8).to(self.rank),
-        )
-
-        ddp_mod = DDP(deepcopy(mod), device_ids=[self.rank])
-        ddp_opt = torch.optim.SGD(ddp_mod.parameters(), lr=0.1)
-        self._test_data_parallel(
-            mod, ddp_mod, opt, ddp_opt, train_batch, train_step, "fully_shard"
-        )
-
-    @skip_if_lt_x_gpu(2)
-    @with_comms
-    def test_fully_shard_sgd_momentum(self):
-        mod = SimpleMLP().cuda(self.rank)
-        opt = torch.optim.SGD(mod.parameters(), lr=0.1, momentum=0.9)
-
-        train_batch = (
-            torch.randn(128, 50).to(self.rank),
-            torch.randn(128, 8).to(self.rank),
-        )
-
-        ddp_mod = DDP(deepcopy(mod), device_ids=[self.rank])
-        ddp_opt = torch.optim.SGD(ddp_mod.parameters(), lr=0.1, momentum=0.9)
-        self._test_data_parallel(
-            mod, ddp_mod, opt, ddp_opt, train_batch, train_step, "fully_shard"
-        )
-
-    @skip_if_lt_x_gpu(2)
-    @with_comms
-    def test_fully_shard_sgd_foreach(self):
-        mod = SimpleMLP().cuda(self.rank)
-        opt = torch.optim.SGD(mod.parameters(), lr=0.1, momentum=0.9, foreach=True)
-
-        train_batch = (
-            torch.randn(128, 50).to(self.rank),
-            torch.randn(128, 8).to(self.rank),
-        )
-
-        ddp_mod = DDP(deepcopy(mod), device_ids=[self.rank])
-        ddp_opt = torch.optim.SGD(
-            ddp_mod.parameters(), lr=0.1, momentum=0.9, foreach=True
-        )
-
-        self._test_data_parallel(
-            mod, ddp_mod, opt, ddp_opt, train_batch, train_step, "fully_shard"
-        )
+        sgd_configs = [
+            {"lr": 0.1},
+            {"lr": 0.1, "momentum": 0.9},
+            {"lr": 0.1, "momentum": 0.9, "foreach": True},
+        ]
+
+        for config in sgd_configs:
```

Comment:
does parametrize work in this case? 

https://github.com/pytorch/pytorch/blob/e2a3817dfd04fbedae804848b451a169168da132/test/distributed/fsdp/test_fsdp_core.py#L122

Reply:
Unfortunately that will make the unit test (much) slower because it will re-initialize process group for each subtest 😢

Reply:
Let's figure out if we could somewhat use parametrize when the configs become larger, we can see if there's someway to avoid re-initialize process group when we try that